### ISSUE TITLE
UI consistency - Make it clearer when a text field is saved in the UI

### ISSUE THREAD

**Issue by franck-boullier** [unee-t/frontend#505]
# The problem:

Today we have several text fields that are editable BUT the UI/UX to edit a text field is not consistent:
## Option 1 - Go to 'Edit Mode':
  - click on an 'Edit' icon 
![image](https://user-images.githubusercontent.com/31331637/45595166-3cde2280-b9da-11e8-9edf-218667f81666.png)
  - Do the change
![image](https://user-images.githubusercontent.com/31331637/45595174-50898900-b9da-11e8-8ca6-2c85d0114871.png)
  - Click on 'Save' when the user is OK with the change or 'Cancel' if the user wants to discard the changes
  - The change is saved only after the user clicks on the 'Save' link.
![image](https://user-images.githubusercontent.com/31331637/45595220-06ed6e00-b9db-11e8-8808-fde1689dd76b.png)

### Pros:
- User has a clear understanding and control of when the field is edited and saved
- User is less likely to unexpectedly edit a field
- We do not send repetitive notifications while the user is changing the field. Notification is sent only when the filed is edited and saved
- This works well if the field already has some text.

### Cons:
- More clicks to edit a field

## Option 2 - 'on the fly' edit are possible

- There is no need to click on anything
![image](https://user-images.githubusercontent.com/31331637/45595238-41efa180-b9db-11e8-81ba-5952bc02b2ae.png)
- Typing something in the field will automatically update the field. No action needed from the user. The filed is 'autosaved'.
![image](https://user-images.githubusercontent.com/31331637/45595255-66e41480-b9db-11e8-916f-4c83edbf4717.png)

### Pros:
- This works better if the field is empty
- Less click to update a field

### Cons:
- User does not know if the field has been saved or not
- Risk to unexpectedly edit a field
- This messes up our notification mechanism. Multiple notifications are sent while the user is changing the field

# A possible Solution:

- If there is already a value in the field always use Option 1.
- If the field is empty
  - No need to click on an 'Edit' icon, it is possible to enter some text in the field
  - Display the Cancel / Save option so we
    - Know when the user is done updating the field
    - Make it clear to the user (albeit implicitly) that his/her update will be saved in the system only if he/she clicks on the 'Save' option
(similar to Option 1)
![image](https://user-images.githubusercontent.com/31331637/45595351-cc84d080-b9dc-11e8-89a5-2d9a2f849e21.png)

@kiatlim, do you see any issue with that idea?
@nbiton do you see any major technical problems with that approacg?

# The Fields that can be updated as of today
## Inspection Report:
- Remarks and Comments
![image](https://user-images.githubusercontent.com/31331637/45595238-41efa180-b9db-11e8-81ba-5952bc02b2ae.png)

## Cases:

These are accessible in the 'More Info/Details' screen for a case ([see an example here](https://case.dev.unee-t.com/case/61926/details))

- Title of the case
![image](https://user-images.githubusercontent.com/31331637/45595380-27b6c300-b9dd-11e8-8c71-aa28bbee1ba6.png)
- Solution
- Next Step
![image](https://user-images.githubusercontent.com/31331637/45595384-3c935680-b9dd-11e8-985f-256761d53e04.png)

**Comment by kiatlim** [comment]
@franck-boullier what about select fields (dropdown options eg. status, priority). Do they remain editable always? (quick access).

So only text fields are having the inline editing option?

**Comment by franck-boullier** [comment]
> what about select fields (dropdown options eg. status, priority). Do they remain editable always? (quick access).

Yes, I think the select fields (dropdown options eg. status, priority) should remain editable always (quick access).

> So only text fields are having the inline editing option?

Correct, the idea I described above should apply only to the text fields

**Comment by kiatlim** [comment]
I'm fine with the idea to always make text fields (regardless of filled up or not filled up) to go into edit mode before the user can edit them. 

I feel that when its not filled up, the user still has to go into edit mode because of consistency, so they don't have to guess when they can edit and when they cannot edit. 

Once the object (case / inventory / IR / unit) is created, text fields must go into edit mode for it to be editable, inline. 

Thoughts?

**Comment by franck-boullier** [comment]
> I feel that when its not filled up, the user still has to go into edit mode because of consistency, so they don't have to guess when they can edit and when they cannot edit.

That works for me. Plus thus is probably faster to implement too...

@nbiton?

**Comment by nbiton** [comment]
@franck-boullier @kiatlim I also think that the solution we used for editing the "Report Title" can be applied to other free text fields on the case info editing form.
We can have empty fields show up as immediately editable, and make the "Cancel" and "Save" buttons show up as the user starts typing.

**Comment by kiatlim** [comment]
@franck-boullier yes its possible to update the designs back in Figma but if there's other things more important we'll look at those first since this would be quite self explanatory for now.

**Comment by franck-boullier** [comment]
For the records:

The improved designs of how we can udpate the "Solution" and "Next Steps" already exists in Figma.
We should keep that in mind when we work on this!

![image](https://user-images.githubusercontent.com/31331637/46056824-21ff7100-c185-11e8-901d-317a8a756f6f.png)

and 

![image](https://user-images.githubusercontent.com/31331637/46056836-3cd1e580-c185-11e8-90b2-262934730032.png)

**Comment by nbiton** [comment]
If we implement this, how do we show the existing solution and next steps?

**Comment by franck-boullier** [comment]
> If we implement this, how do we show the existing solution and next steps?

We used the latest version of the design as shown in [figma](https://www.figma.com/file/SgLcXdmBih1JxVq1lupMiPtr/Unee-T-Designs?node-id=2593%3A4923): 

![image](https://user-images.githubusercontent.com/31331637/59322450-bb1cea80-8d08-11e9-8c0f-685e07b4490c.png)

The "Pencil" icon opens the popup and the full text of the solution is pre-selected
![image](https://user-images.githubusercontent.com/31331637/59322498-ef90a680-8d08-11e9-8994-a97873e0379b.png)

**Comment by katelyp** [comment]
We have discussed this over today's meeting. Notes are on this [figma screen](https://www.figma.com/file/SgLcXdmBih1JxVq1lupMiPtr/Unee-T-Designs?node-id=0%3A2876)

Will follow up with screens and links after redesign
